database/sqlite/sqlite_metadata.c

```diff
@@ -0,0 +1,1485 @@
+                    break;
+                case METADATA_ADD_HOST_INFO:
+                    dict_item = (DICTIONARY_ITEM * ) cmd.param[0];
+                    host = (RRDHOST *) dictionary_acquired_item_value(dict_item);
+                    rc = sql_store_host_info(host);
+                    if (unlikely(rc))
+                        error_report("Failed to store host info in the database for %s", string2str(host->hostname));
+                    dictionary_acquired_item_release(rrdhost_root_index, dict_item);
+                    break;
+                case METADATA_STORE_CLAIM_ID:
+                    store_claim_id((uuid_t *) cmd.param[0], (uuid_t *) cmd.param[1]);
+                    freez((void *) cmd.param[0]);
+                    freez((void *) cmd.param[1]);
+                    break;
+                case METADATA_STORE_HOST_LABELS:
+                    dict_item = (DICTIONARY_ITEM * ) cmd.param[0];
+                    host = (RRDHOST *) dictionary_acquired_item_value(dict_item);
+                    rc = exec_statement_with_uuid(SQL_DELETE_HOST_LABELS, &host->host_uuid);
+
+                    if (likely(rc == SQLITE_OK)) {
+                        buffer_flush(work_buffer);
+                        struct query_build tmp = {.sql = work_buffer, .count = 0};
+                        uuid_unparse_lower(host->host_uuid, tmp.uuid_str);
+                        rrdlabels_walkthrough_read(host->rrdlabels, host_label_store_to_sql_callback, &tmp);
+                        db_execute(buffer_tostring(work_buffer));
+                    }
+
+                    dictionary_acquired_item_release(rrdhost_root_index, dict_item);
+                    break;
+
+                case METADATA_SCAN_HOSTS:
+                    if (unlikely(metadata_flag_check(wc, METADATA_FLAG_SCANNING_HOSTS)))
+                        break;
+
+                    struct scan_metadata_payload *data = mallocz(sizeof(*data));
+                    data->wc = wc;
+                    data->completion = cmd.completion;  // Completion by the worker
+
+                    if (unlikely(cmd.completion)) {
+                        data->max_count = 0;            // 0 will process all pending updates
+                        cmd.completion = NULL;          // Do not complete after launching worker (worker will do)
+                    }
+                    else
+                        data->max_count = 1000;
+
+                    worker_request.data = data;
+                    metadata_flag_set(wc, METADATA_FLAG_SCANNING_HOSTS);
+                    if (unlikely(
+                            uv_queue_work(loop,&worker_request,
+                                          start_metadata_hosts,
+                                          after_metadata_hosts))) {
+                        // Failed to launch worker -- let the event loop handle completion
+                        cmd.completion = data->completion;
+                        freez(data);
```

Comment:
Considering the `freez` inside a conditional, I suppose the function will never fail, but if it will never fail, why do we need a `if`? :thinking:

Reply:
It could fail. If it doesn't fail then the freez will happen in the `after_metadata_hosts` function